pkg/controller/instanceset/instance_util.go

```diff
@@ -173,6 +174,47 @@ func isHealthy(pod *corev1.Pod) bool {
 	return isRunningAndReady(pod) && !isTerminating(pod)
 }
 
+// isRoleReady returns true if pod has role label
+func isRoleReady(pod *corev1.Pod, roles []workloads.ReplicaRole) bool {
+	if len(roles) == 0 {
+		return true
+	}
+	_, ok := pod.Labels[constant.RoleLabelKey]
+	return ok
+}
+
+// isImageMatched returns true if all container statuses have same image as defined in pod spec
+func isImageMatched(pod *corev1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		index := slices.IndexFunc(pod.Status.ContainerStatuses, func(status corev1.ContainerStatus) bool {
+			return status.Name == container.Name
+		})
+		if index == -1 {
+			continue
+		}
+		specImage := container.Image
+		statusImage := pod.Status.ContainerStatuses[index].Image
+		// Image in status may not match the image used in the PodSpec.
+		// More info: https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodStatus
+		specName, specTag, specDigest := image.Split(specImage)
+		statusName, statusTag, statusDigest := image.Split(statusImage)
+		if len(specDigest) != 0 && specDigest != statusDigest {
+			return false
+		}
+		if len(specTag) != 0 && specTag != statusTag {
+			return false
+		}
+		specSlashIndex := strings.Index(specName, "/")
+		if specSlashIndex != -1 && specName != statusName {
```

Comment:
Is this logic for comparing "foo" and "docker.io/library/foo"? Please add a comment to explain.

Reply:
Yes. fixed by [4a830d9](https://github.com/apecloud/kubeblocks/pull/8385/commits/4a830d90058110c8a368e51cb6565a5600ee2867)